pkg/asset/logging/filelogging_test.go

```diff
@@ -0,0 +1,145 @@
+		},
+		{
+			name: "test asset with one file",
+			assets: []asset.WritableAsset{
+				&installconfig.InstallConfig{
+					File: &asset.File{
+						Filename: "a.yaml",
+					},
+				},
+			},
+			cmdName:               "test install config",
+			directory:             "test",
+			expectedGenerationLog: "test install config created in: test",
+		},
+		{
+			name: "test asset with two files, same directory",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test",
+		},
+		{
+			name: "test asset with two files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test and test/machines",
+		},
+		{
+			name: "test asset with two files, two directories, but three entries (same file twice",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test and test/machines",
+		},
+		{
+			name: "test asset with three files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test/",
+			expectedGenerationLog: "machines created in: test and test/machines",
+		},
+		{
+			name: "test asset with three files, three directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("control-plane/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test, test/control-plane and test/machines",
+		},
+		{
+			name: "test asset with five files, five directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("control-plane/c.yaml"),
+						getAsset("master/c.yaml"),
+						getAsset("worker/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test, test/control-plane and test/machines",
```

Comment:
why did you choose 2 of 4 sub directories. summarize to the top-level when number of directories to be listed is greater than the const max.

Reply:
Since the user specifies the directory where the files should go, I thought the message (target) created in (specified directory) is not that informational. But, specifying just two of the directories would be misleading so removing.